### PR TITLE
Fixes Pulp w/ Mongoengine v0.14.1+

### PR DESCRIPTION
**Pull request by dralley**
closes #3687
https://pulp.plan.io/issues/3687

### CODE REVIEW

File: server/pulp/server/db/model/__init__.py [hunk 1]
```diff
@@ -553,8 +553,14 @@ def save_with_set_on_insert(self, fields_to_set_on_insert):
         set_on_insert = {}
         for field in fields_to_set_on_insert:
             set_on_insert[field] = stuff_to_update.pop(field)
+
         task_id = stuff_to_update.pop('task_id')
 
+        # Don't try to set Mongoengine internal values.
+        # Can cause silent failures if you accidentally overwrite 'id'.
+        for internal_field in ('id', '_ns'):
```
Comment (goosemania) [inline server/pulp/server/db/model/__init__.py]: How did you choose which fields are internal? E.g. `_id` is not in the tuple.

Comment (asmacdo) [reply server/pulp/server/db/model/__init__.py]: @goosemania Mongoengine uses `id` as an alias for `_id`

Comment (goosemania) [reply server/pulp/server/db/model/__init__.py]: @asmacdo , thank you

**Comment by goosemania** [comment]
@dralley , thanks for the fix! :tada: 
It works for me both with the mongoengine prior to 0.14.1 and with 0.15.

File: server/pulp/server/db/model/__init__.py (hunk 1 above)
Comment (dralley) [reply server/pulp/server/db/model/__init__.py]: @goosemania exactly, and as far as methods go, I interactively debugged the _data object to determine what was being added to it.

**Pull request merged**